ddl/ddl_api.go

```diff
@@ -1815,6 +1804,9 @@ func (d *ddl) CreateTable(ctx sessionctx.Context, s *ast.CreateTableStmt) (err e
 	if err != nil {
 		return errors.Trace(err)
 	}
+	if err = setTemporaryType(ctx, tbInfo, s); err != nil {
+		return errors.Trace(err)
+	}
```

Comment:
You can't put it here. `buildTableInfoWithStmt` relies on `tbInfo.TempTableType`. 
For example, previously, `buildTableInfoWithStmt` sets `tbInfo.TempTableType` and then calls `handleTableOptions`. But `handleTableOptions` has such codes:
```go
    if tbInfo.TempTableType != model.TempTableNone {
```